### PR TITLE
[Immediate] JIT'ing Globals

### PR DESCRIPTION
**Pull request by zachary-kent**
This PR adds support for emitting global variables on demand in lazy immediate mode by extending the `SymbolSource` class.

Depends on #67953

### CODE REVIEW

File: include/swift/SIL/SILModule.h [hunk 1]
```diff
@@ -737,6 +737,10 @@ class SILModule {
     return isPossiblyUsedExternally(getDeclSILLinkage(decl), isWholeModule());
   }
 
+  /// Promote the linkage of every entity in this SIL module so that they are
+  /// externally visible.
```
Comment (slavapestov) [inline include/swift/SIL/SILModule.h:741]: Can you add a comment explaining what this is used for?

File: test/Interpreter/lazy/deferred_syntax_errors.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// REQUIRES: OS=macosx
+// RUN: %target-jit-run %s -enable-experimental-feature LazyImmediate | %FileCheck %s
+
+// Tests that parsing of function bodies is deferred
```
Comment (slavapestov) [inline test/Interpreter/lazy/deferred_syntax_errors.swift:4]: Nice!

**Comment by zachary0kent** [comment]
@swift-ci please test

**Comment by zachary0kent** [comment]
@swift-ci please test

**Pull request merged**